### DISCUSSION TITLE
torch_geometric.explain and edge features

### DISCUSSION THREAD

**Discussion by arthurserres** [pyg-team/pytorch_geometric#8539] [General]
Hello,
 I am trying to explain my GNN using the explain experimental module. All the features that I use in my model are on the edges and not on the nodes. I thus wanted to consider the mask "object" for node and "common_attributes" which is the correct configuration If I goodly understood the use of masks. However I looked at the source code and understand that edge mask can only be None or "object", so if I understand right, edge attribute cannot be used to explain the outcomes, is that correct ? 
Thanks in advance.

**Comment by rusty1s** [comment]
Yes, this is correct. While it is not impossible to support explanations of edge features, it makes the implementation a bit harder since we now need to separate the implementation of explainers based on the `edge_mask` property, i.e., `edge_mask=object` refers to explanation of `edge_index`, while `edge_mask=common_attributes` refers to explanation of `edge_attr`.

**Reply by arthurserres** [reply]
I see, thanks a lot for the update it's nice of you!
 I saw that Captum can [natively be used to explain GNN](https://colab.research.google.com/drive/1fLJbFPz0yMCQg81DdCP5I8jXw9LoggKO?usp=sharing) (I think it essentialy does what we can do with the captum explainer and the explain class, correct ? ). 
Is there a chance I can easily modify this notebook or the source code of explain such that it also produces explaination using edge features ? Or is it a more complicated thing that requires high level coding skills ?

Thanks by advance :)

**Reply by rusty1s** [reply]
What we need to change is the [`to_captum_input`](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/explain/algorithm/captum.py#L179) function, which also respects `edge_attr` as an input feature and includes it into its attribution.

**Reply by arthurserres** [reply]
Thanks a lot, I will have a look :)
